src/libOpenImageIO/imagebufalgo_opencv.cpp

```diff
@@ -136,7 +137,161 @@ IplImage *
 ImageBufAlgo::to_IplImage (const ImageBuf &src)
 {
 #ifdef USE_OPENCV
-    return NULL;
+    ImageSpec spec = src.spec();
+    ImageBuf tmp = src;
+
+    // If the image has 4+ channels, then reduce to 4 channels, and use BGRA
+    // order (OpenCV ordering).
+    if (spec.nchannels >=3) {
+        int channelorder[spec.nchannels];
+
+        // Set the channel order
+        for (int i = 0; i < spec.nchannels; ++i) {
+            channelorder[i] = i;
+        }
+
```

Comment:
I find the whitespace helps me chunk the code into blocks of logic. I can trim this if you prefer, though.